apps/sms/views/shared/test/unit/contacts_test.js

```diff
@@ -150,111 +38,82 @@ suite('Contacts', function(done) {
     window.fb = mockReaderUtils;
   });
 
-  teardown(function() {
-    navigator.mozContacts.mHistory.length = 0;
-  });
-
   suiteTeardown(function() {
     navigator.mozContacts = nativeMozContacts;
     window.fb = realFb;
+    //Restoring the native API
+    navigator.mozContacts.find.restore();
   });
 
   suite('Contacts.findContactByString, single-term', function() {
 
     test('(string[tel,givenName,familyName], ...) Match', function(done) {
-      var mozContacts = navigator.mozContacts;
+      var result = MockContact.list();
 
-      Contacts.findContactByString('O\'Hare').then(function(contacts) {
-        var mHistory = mozContacts.mHistory;
+      navigator.mozContacts.find.returns(Promise.resolve(result));
```

Comment:
to be sure that `findContactByString` calls the API correctly, maybe we can use `withArgs` here:

```javascript
var name = 'O\'Hare';
navigator.mozContacts.find.withArgs(name).returns(
  Promise.resolve(MockContact.list())
); 
Contacts.findContactByString(name).then(function(contacts) {
 ....
```

What do you think ?